x-pack/plugins/security/server/lib/authorization/privilege_serializer.ts

```diff
@@ -6,73 +6,73 @@
 
 const featurePrefix = 'feature_';
 const spacePrefix = 'space_';
-const reservedPrivilegeNames = ['all', 'read'];
+const minimumPrivilegeNames = ['all', 'read'];
+const spaceMinimumPrivilegeNames = minimumPrivilegeNames.map(
+  privilegeName => `${spacePrefix}${privilegeName}`
+);
+const deserializeFeaturePrivilegeRegexp = new RegExp(
+  `^${featurePrefix}([a-zA-Z0-9_-]+)\\.([a-zA-Z0-9_-]+)$`
+);
+
+interface FeaturePrivilege {
+  featureId: string;
+  privilege: string;
+}
 
 export class PrivilegeSerializer {
-  public static serializeGlobalReservedPrivilege(privilegeName: string) {
-    if (!reservedPrivilegeNames.includes(privilegeName)) {
-      throw new Error('Unrecognized global reserved privilege');
+  public static isGlobalMinimumPrivilege(privilegeName: string) {
+    return minimumPrivilegeNames.includes(privilegeName);
+  }
+
+  public static isSpaceMinimumPrivilege(privilegeName: string) {
+    return spaceMinimumPrivilegeNames.includes(privilegeName);
+  }
+
+  public static serializeGlobalMinimumPrivilege(privilegeName: string) {
+    if (!minimumPrivilegeNames.includes(privilegeName)) {
```

Comment:
nit: what do you think about using `isGlobalMinimumPrivilege` here, instead of duplicating the admittedly small amount of logic?

Reply:
I've gone back and forth on this a few times... I actually need to go back and rename `isSpaceMinimumPrivilege` to `isSerializedSpaceMinimumPrivilege`, as that's the real intent. If that's the case, I think that we should also change `isGlobalMinimumPrivilege` to `isSerializedGlobalMinimumPrivilege`, and if that's the case it just so happens that the serialized and unserialized format for global minimum privileges are the same values, and I don't think that we should be reusing `isSerializedGlobalMinimumPrivilege` when actually serializing the global minimum privilege 🤷‍♂️

Reply:
Thanks for explaining - I agree we shouldn't use `isSerialized*Privilege` within the serialized functions, so what you have here is good with me